src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForTopContainer.cs

```diff
@@ -33,5 +40,11 @@ public override IContainer CreateNewEntity(MoBiSpatialStructure spatialStructure
          newEntity.ContainerType = ContainerType.Organism;
          return newEntity;
       }
+
+      protected override IMoBiCommand AddNeighborhoodsToSpatialStructure(IReadOnlyList<NeighborhoodBuilder> neighborhoods, MoBiSpatialStructure spatialStructure)
+      {
+         //we need to make sure that we use the correct interaction task for the spatial structure in order to add the neighborhoods to the top container
+         return _interactionTaskForNeighborhood.AddTo(neighborhoods, spatialStructure.NeighborhoodsContainer, spatialStructure);
```

Comment:
This is how I managed to call a task from another task sharing the same base class where the code is actually implemented lol. abstract method in base that calls specific implementation. One class references the other and the other simply calls itself. A bit of a weird way but elegant at the end I think